universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx

```diff
@@ -25,11 +25,19 @@ export interface ChooseTopUpMethodProps {
 }
 
 export const ChooseTopUpMethod = ({sdk, contractAddress, onPayClick, topUpClassName, logoColor, isDeployment, onGasParametersChanged}: ChooseTopUpMethodProps) => {
+  const [gasParameters, setGasParameters] = useState<GasParameters | undefined>(undefined);
   if (isDeployment) {
     ensureNotNull(onGasParametersChanged, MissingParameter, 'onGasParametersChanged');
   }
+  const gasParametersChanged = (gasParameters: GasParameters) => {
+    if (isDeployment) {
+      setGasParameters(gasParameters);
+      onGasParametersChanged!(gasParameters);
+    }
+  };
```

Comment:
Do we need this check if `gasParametersChanged` is used only in GasPrice component which is shown only if isDeployment === true